### PR TITLE
Conway Shim API

### PR DESCRIPTION
**Pull request by nickcastel50**
This is a full implementation of a shim for the Web-IFC read API capabilities. I will place build instructions here since I don't want to edit the developer guide just yet until it lands, and there are currently no extra build instructions in the readme.

## Build Instructions:

1. ```yarn add ./bldrs-conway-v0.0.1.tgz --production```
2. ```yarn add shx```

### Conway Shim:

1. ```yarn build-conway```
2. ```yarn serve-share-conway```

### Original Web-IFC

1. ```yarn build-webifc```
2. ```yarn serve-share-webifc```

### CODE REVIEW

**Comment by pablo-mayrgundter** [comment]
Per convo, logging and error should accumulate to a vector during Init, OpenModel and Geometry/Mesh calls, etc. to optionally be accessed after each call by a method compatible with web-ifc, [GetAndClearErrors](https://ifcjs.github.io/web-ifc/docs/classes/web_ifc.IfcAPI.html#GetAndClearErrors).

Only variant we thought up was to keep a map by IFC type, or perhaps by error type, to keep counts instead of 1:1 vector, since there is usually a high count involved that is slow to log.

**Comment by ConorStokes** [comment]
This model is probably not appropriate for the server-side use cases where
a process might fall over halfway through (say due to hitting the heap
allocation limit) and we'd want the contents of the log to determine where.

I strongly recommend abstracting logging such that accumulating in a vector
in this fashion is optional for use cases where it's appropriate. That's
partly why I suggested using a proxy that allowed multiple logging targets.

**Comment by pablo-mayrgundter** [comment]
Ok, builds for me.. nice :)

Please respond to My&Conor's comments about logging.

Couples questions on build.   Where did you land with Ogali's build process?  Should we see 2 previews here?  Also, is this safe to check-in and won't affect production?

**Comment by nickcastel50** [comment]
> Ok, builds for me.. nice :)
> 
> Please respond to My&Conor's comments about logging.
> 
> Couples questions on build. Where did you land with Ogali's build process? Should we see 2 previews here? Also, is this safe to check-in and won't affect production?

Regarding logging, a logging proxy that maps to multiple targets makes sense. I think this should come in a later PR though as geometry type coverage is taking precedence, and then we can circle back and hit logging across Conway / Conway-Geom, Shim, Sentry, etc. 


As for build process, it looks like it's just generating one link still. If you checked this in, it would currently deploy Conway Shim and not Web-IFC, which might not be what we want yet due to coverage. We should page @oogali about how we can deploy this using the subdomain(s) as discussed.

**Comment by pablo-mayrgundter** [comment]
Ok yep, should check in without overriding prod.. LGTM and I'll hand off to Ogali now

**Comment by pablo-mayrgundter** [comment]
image

**Comment by pablo-mayrgundter** [comment]
Hey Nick,

We brought up at mtg the possibility that all you need to do here is flip the default from conway to web-ifc.  Is that all that's left to determine?

If so, I suggest trying that and submitting, but stand by to revert if need be.

How's the sound?

**Comment by nickcastel50** [comment]
> Hey Nick,
> 
> We brought up at mtg the possibility that all you need to do here is flip the default from conway to web-ifc. Is that all that's left to determine?
> 
> If so, I suggest trying that and submitting, but stand by to revert if need be.
> 
> How's the sound?

Hey @pablo-mayrgundter, Ogali and I took a look on Friday. In addition to changing the default "build" back to web-ifc (that change is already here), there are a few changes I just pushed up for Github actions, so it will actually test and run both the shim and original web-ifc backend. Once I can confirm the Github Action changes work we should be good to go.

File: package.json [hunk 1]
```diff
@@ -42,6 +47,7 @@
     "@octokit/rest": "^19.0.3",
     "@sentry/react": "^7.61.0",
     "@sentry/tracing": "^7.61.0",
+    "bldrs-conway": "./bldrs-conway-v0.0.1.tgz",
```
[the remark this answers is not in the document]
Comment (nickcastel50) [reply package.json:50]: @oo-bldrs that is accomplished by the following.

In conway repo root:


```
yarn build-all-release-web
yarn pack --production
```

Then the resulting _bldrs-conway-v0.0.1.tgz_ can be copied to the share root.

**Pull request merged**